aptos-move/framework/aptos-names/sources/verify.move

```diff
@@ -0,0 +1,35 @@
+module aptos_names::verify {
+    use std::string;
+    use std::vector;
+    use aptos_framework::chain_id;
+    use aptos_std::ed25519;
+    use aptos_names::config;
+
+    friend aptos_names::domains;
+
+    struct RegisterDomainProofChallenge has drop {
+        sequence_number: u64,
+        register_address: address,
+        domain_name: string::String,
+        chain_id: u8,
+    }
+
+    const EINVALID_PROOF_OF_KNOWLEDGE: u64 = 1;
+
+    public(friend) fun verify_register_domain_signature(signature: vector<u8>, sequence_number: u64, account_address: address, domain_name: string::String) {
```

Comment:
This is a nitpick, but not really 😅: This API should be called `assert_register_domain_signature_verifies` to more clearly indicate to the caller that this function will abort if the signature doesn't verify.

In other words, **please, please, please** 😄, let's not form a habit of ignoring return values of functions that start with `verify_*` (e.g., like `ed25519::verify_signature_strict_t`). That would be a very, very bad habit...

Reply:
okay makes sense! just updated the function name : )